src/game_actor.cpp

```diff
@@ -119,7 +119,7 @@ bool Game_Actor::IsSkillUsable(int skill_id) const {
 		return false;
 	}
 	for (int inflictedState : this->GetInflictedStates()) {
-		if ((Data::states[inflictedState - 1].restrict_magic && Data::skills[skill_id].magical_rate > Data::states[inflictedState - 1].restrict_magic_level) || ((Data::states[inflictedState - 1].restrict_skill && Data::skills[skill_id-1].hit > Data::states[inflictedState - 1].restrict_skill_level))) {
+		if ((Data::states[inflictedState - 1].restrict_magic && Data::skills[skill_id - 1].magical_rate > Data::states[inflictedState - 1].restrict_magic_level) || ((Data::states[inflictedState - 1].restrict_skill && Data::skills[skill_id-1].hit > Data::states[inflictedState - 1].restrict_skill_level))) {
```

Comment:
This code is already implemented in Game_Battler::IsSkillUsable. So I was wrong when I said that silence is not working yet. Sorry :(